Sources/Helpers/DownloadingListHelper/DownloadingListHelper.swift

```diff
@@ -0,0 +1,134 @@
+    private var downloadTaskCompletedObserver: OAAutoObserverProxy?
+    private var localResourcesChangedObserver: OAAutoObserverProxy?
+    private var downloadTaskCount = 0
+    
+    override init() {
+        downloadsManager = OsmAndApp.swiftInstance().downloadsManager
+        super.init()
+        downloadTaskProgressObserver = OAAutoObserverProxy(self, withHandler: #selector(onDownloadResourceTaskProgressChanged), andObserve: OsmAndApp.swiftInstance().downloadsManager.progressCompletedObservable)
+        downloadTaskCompletedObserver = OAAutoObserverProxy(self, withHandler: #selector(onDownloadResourceTaskFinished), andObserve: OsmAndApp.swiftInstance().downloadsManager.completedObservable)
+        localResourcesChangedObserver = OAAutoObserverProxy(self, withHandler: #selector(onLocalResourcesChanged), andObserve: OsmAndApp.swiftInstance().localResourcesChangedObservable)
+    }
+    
+    func hasDownloads() -> Bool {
+        return !getDownloadingTasks().isEmpty
+    }
+    
+    func getDownloadingTasks() -> [OADownloadTask] {
+        guard let keys = downloadsManager.keysOfDownloadTasks() else { return [] }
+        var tasks = [OADownloadTask]()
+        for key in keys {
+            if let task = downloadsManager.firstDownloadTasks(withKey: key as? String) {
+                tasks.append(task)
+            }
+        }
+        tasks.sort { $0.name < $1.name }
+        return tasks
+    }
+    
+    func buildAllDownloadingsCell() -> OATitleIconProgressbarCell? {
+        var cell: OATitleIconProgressbarCell?
+        if let allDownloadingsCell {
+            cell = allDownloadingsCell
+        } else {
+            let nib = Bundle.main.loadNibNamed(OATitleIconProgressbarCell.reuseIdentifier, owner: self, options: nil)
+            cell = nib?.first as? OATitleIconProgressbarCell
+        }
+        guard let cell else { return nil }
+        
+        cell.accessoryType = .disclosureIndicator
+        cell.imgView.image = UIImage.templateImageNamed("ic_custom_multi_download")
+        cell.imgView.tintColor = UIColor.iconColorActive
+        
+        var title = localizedString("downloading") + ":"
+        let tasks = getDownloadingTasks()
+        for i in 0 ..< tasks.count {
+            if !tasks[i].name.isEmpty {
+                title += " " + tasks[i].name
+                if i != tasks.count - 1 {
+                    title += ","
+                }
+            }
+        }
+        cell.textView.text = title
+        
+        cell.progressBar.setProgress(Float(calculateAllDownloadingsCellProgress()), animated: false)
+        cell.progressBar.progressTintColor = UIColor.iconColorActive
+        
+        allDownloadingsCell = cell
+        return allDownloadingsCell
+    }
+    
+    func getListViewController() -> DownloadingListViewController {
+        let vc = DownloadingListViewController()
+        weak var weakSelf = self
+        vc.delegate = weakSelf
```

Comment:
Why do we pass weakSelf to the delegate? The weak var delegate: DownloadingCellResourceHelperDelegate? already has a weak reference; it seems like we can just use self